funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
         <th>Actions</th>
       </tr>
     </thead>
-    <tbody>
-      {%- for p in participants %}
-        <tr id="p-{{ p.id }}">
-          <td class='js-searchable'>{{ p.fullname }}</td>
-          <td class='js-searchable'>{{ p.email }}</td>
-          <td class='js-searchable'>{{ p.company }}</td>
+    <tbody id="participants-table-content">
+      {% raw %}
+      <script id='participant-row' type='text/ractive'>
+      {{#each participants}}
+        <tr id="p-{{ pid }}">
+          <td class='js-searchable'>{{ fullname }}</td>
+          <td class='js-searchable'>{{ email }}</td>
+          <td class='js-searchable'>{{ company }}</td>
           <td>
             <ul class="list-inline status-btn-list">
             <li>
-              <a href={{ "{0}participant/{1}/badge".format(space.url_for(), p.id) }} class={% if p.badge_printed %} 'btn btn-xs btn-success'  {% else %} 'btn btn-xs btn-info' {% endif %} target="_blank">Show badge</a>
+              <a href="{{ showbadge_url }}" {{#badge_printed}} class="btn btn-xs btn-success" {{/badge_printed}} {{^badge_printed}} class="btn btn-xs btn-info" {{/badge_printed}} target="_blank">Show badge</a>
             </li>
             <li>
-              <a href={{ "{0}participant/{1}/edit".format(space.url_for(), p.id) }} class='btn btn-xs btn-default'>Edit</a>
+              <a href="{{ edit_url }}" class="btn btn-xs btn-default" target="_blank">Edit</a>
             </li>
             <li>
-            {%- if p.checked_in %}
-              <a href="{{ event.name }}/checkin/{{ p.id }}" class='btn btn-xs btn-danger'>Cancel Check-in</a>
-            {%- else %}
-            <a href="{{ event.name }}/checkin/{{ p.id }}?checkin=t" class='btn btn-xs btn-warning'>Check-in</a>
-            {%- endif %}
+            {{#checked_in}}
+              <a href="{{ checkin_url }}/checkin/{{ pid }}" data-pid="{{ pid }}" class="btn btn-xs js-cancel-checkin">Cancel Check-in</a>
+              <i class="fa fa-spinner fa-pulse loader"></i>
+              <i class="fa fa-close loader js-abort"></i>
+            {{/checked_in}}
+            {{^checked_in}}
+              <a href="{{ checkin_url }}/checkin/{{ pid }}?checkin=t" data-pid="{{ pid }}"sd class="btn btn-xs js-checkin">Check-in</a>
+              <i class="fa fa-spinner fa-pulse loader"></i>
+              <i class="fa fa-close loader js-abort"></i>
+            {{/checked_in}}
             </li>
             </ul>
           </td>
         </tr>
-      {%- endfor %}
+      {{/each}}
+      </script>
+      {% endraw %}
     </tbody>
   </table>
 {% endblock %}
 {% block footerscripts %}
 <script>
+  if (Modernizr.localstorage) {
+    window.Store = {};
```

Comment:
Store and Queue are explicitly defined as global to avoid being overwritten. At this point of code, var Store is same as window.Store.